src/main/java/com/mathworks/ci/MatlabBuilder.java

```diff
@@ -271,6 +280,66 @@ public FormValidation doCheckTaCoberturaChkBx(@QueryParameter boolean taCobertur
             }
             
 
+            return FormValidation.ok();
+        };
+        
+        public FormValidation doCheckTaModelCoverageChkBx(@QueryParameter boolean taModelCoverageChkBx) {
+            List<Function<String, FormValidation>> listOfCheckMethods =
+            new ArrayList<Function<String, FormValidation>>();
+            final String matlabRoot = Jenkins.getInstance()
+            .getDescriptorByType(MatlabDescriptor.class).getMatlabRoot();
+            if (taModelCoverageChkBx) {
+                listOfCheckMethods.add(chkModelCoverageSupport);
+            }
+            return Jenkins.getInstance().getDescriptorByType(MatlabDescriptor.class)
+            .getFirstErrorOrWarning(listOfCheckMethods, matlabRoot);
+        }
+        
+        Function<String, FormValidation> chkModelCoverageSupport = (String matlabRoot) -> {
+            rel = new MatlabReleaseInfo(matlabRoot);
+            final MatrixPatternResolver resolver = new MatrixPatternResolver(matlabRoot);
+            if(!resolver.hasVariablePattern()) {
```

Comment:
I'm not entirely familiar with how this resolver works, but what happens if the resolver does have a variable pattern? Can you elaborate?

Reply:
If the resolver has variable pattern we don't throw any error or warning on the UI.
 
This is with respect to MATRIX build wherein user can specify variable (C:\Program Files\MATLAB\**$VER**) to determine different versions in which he/she wants to run their tests. To handle this case we don't throw any error or warning on the UI and user gets to see the error/warning at the console output during run-time.